### ISSUE TITLE
Immediate back button after "start/join call" results in broken state

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/talk-android#724]
### Steps
1. Have a conversation with a slow internet connection
2. Go to a conversation
3. Press "Start call"
4. Immediately press back

### Expected
If the join call request finishes, it should be left again afterwards

### Actually
"Leave call" reaches the server before the join finishes.
This leads to wrong order of join/leave messages and results in "has_call" being left on for the conversation. Of course the API should make sure to clear up the state when a person leaves, but then again the requests should also be sent in the correct order (dont leave before join finished)

> ![Bildschirmfoto von 2020-02-11 15-09-14](https://user-images.githubusercontent.com/213943/74244083-2f1cf180-4ce1-11ea-8e00-ceae22bd8dc9.png)

**Comment by mario** [comment]
Yes, requests ordering is tricky business unless you limit the number of concurrent connections to 1, which would make it all the more fun. But I think I know how to solve this and will work on it in the future.

**Comment by nickvergessen** [comment]
> Yes, requests ordering is tricky business unless you limit the number of concurrent connections to 1

Well you could also store the state somewhere and when you finally get the response of the "join" you check if the user wants to leave already and actually only send the "leave" then.

**Comment by mario** [comment]
Yup, being worked on already :)

**Comment by ctilluma** [comment]
So if the back button is pushed and we have already sent out a join request, we need to wait for a response or timeout and then send the terminate message, and put that in an async task so that we don't affect the UI for the user who hit back and is already trying to do something else.
